src/psbt.cpp

```diff
@@ -361,7 +361,7 @@ bool SignPSBTInput(const SigningProvider& provider, PartiallySignedTransaction&
     sigdata.witness = false;
     bool sig_complete;
     if (txdata == nullptr) {
-        sig_complete = ProduceSignature(provider, DUMMY_SIGNATURE_CREATOR, utxo.scriptPubKey, sigdata);
+        sig_complete = ProduceSignature(provider, grind ? DUMMY_SIGNATURE_CREATOR : DUMMY_MAXIMUM_SIGNATURE_CREATOR, utxo.scriptPubKey, sigdata);
```

Comment:
It seems like the dummy signature produced here isn't used for fee estimation, but for figuring out what signatures that the psbt is missing (for use when analyzing a psbt). It looks like this behavior was introduced in cb40b3abd4514361a024a1e7a1a281da9261261b.

Reply:
@achow101 I could use some hints here.

Reply:
Size estimation occurs in `CreateTransaction`, inside of `SelectCoins` and `CalculateMaximumSignedInputSize`. `FillPSBT` and the functions it calls shouldn't be touched.